homeassistant/components/amcrest/binary_sensor.py

```diff
@@ -211,13 +211,16 @@ async def _async_update_others(self) -> None:
             log_update_error(_LOGGER, "update", self.name, "binary sensor", error)
             return
 
-        if (event_code := self.entity_description.event_code) is None:
+        if (event_codes := self.entity_description.event_codes) is None or len(
+            event_codes
+        ) == 0:
             _LOGGER.error("Binary sensor %s event code not set", self.name)
             return
 
         try:
-            self._attr_is_on = (
+            self._attr_is_on = any(  # type: ignore[arg-type]
```

Comment:
Why do we need to ignore type here?

Reply:
I think there is some problem with mypy being able to reason through the types with the async expression inside the generator. Without this, I get:
```
error: Argument 1 to "any" has incompatible type "AsyncGenerator[bool, None]"; expected "Iterable[object]"  [arg-type]
```
I can change the generator to a list comprehension and then mypy is happy without the `type: ignore` here, however, that upsets pylint. I figured it would be better to use the generator and silence mypy than to use the list comprehension and silence pylint.

Reply:
Ok.